www/source/blog/2017-08-09-Habitat-plans-for-net-full-iis-apps.md

```diff
@@ -0,0 +1,301 @@
+  "core/nuget",
+  "core/dotnet-47-dev-pack"
+)
+
+function invoke-download { }
+function invoke-verify { }
+
+function Invoke-Build {
+  Copy-Item $PLAN_CONTEXT/../../* $HAB_CACHE_SRC_PATH/$pkg_dirname -recurse -force
+  nuget restore $HAB_CACHE_SRC_PATH/$pkg_dirname/$pkg_name/packages.config -PackagesDirectory $HAB_CACHE_SRC_PATH/$pkg_dirname/packages -Source "https://www.nuget.org/api/v2"
+  nuget install MSBuild.Microsoft.VisualStudio.Web.targets -Version 14.0.0.3 -OutputDirectory $HAB_CACHE_SRC_PATH/$pkg_dirname/
+  $env:TargetFrameworkRootPath="$(Get-HabPackagePath dotnet-47-dev-pack)\Program Files\Reference Assemblies\Microsoft\Framework"
+  $env:VSToolsPath = "$HAB_CACHE_SRC_PATH/$pkg_dirname/MSBuild.Microsoft.VisualStudio.Web.targets.14.0.0.3/tools/VSToolsPath"
+  ."$env:SystemRoot\Microsoft.NET\Framework64\v4.0.30319\MSBuild.exe" $HAB_CACHE_SRC_PATH/$pkg_dirname/$pkg_name/${pkg_name}.csproj /t:Build /p:VisualStudioVersion=14.0
+  if($LASTEXITCODE -ne 0) {
+      Write-Error "dotnet build failed!"
+  }
+}
+
+function Invoke-Install {
+  ."$env:SystemRoot\Microsoft.NET\Framework64\v4.0.30319\MSBuild.exe" $HAB_CACHE_SRC_PATH/$pkg_dirname/$pkg_name/${pkg_name}.csproj /t:WebPublish /p:WebPublishMethod=FileSystem /p:publishUrl=$pkg_prefix/www
+}
+```
+
+This bit of powershell script essentially leverages `nuget` and `msbuild` to grab dependencies, compile our code and publish it to our habitat `pkgs` directory which will be archived to a Habitat `hart` package. It should also be noted that in order for Habitat to build this plan, one does NOT need to have Visual Studio installed. Rather you just need a recent framework version of .NET installed which ships by default on server 2012R2 and forward. 
+
+Lets build and package our `hart` for this application:
+
+```
+C:\dev\hab-sln> hab studio enter -w
+   hab-studio: Creating Studio at /hab/studios/dev--hab-sln
+» Importing origin key from standard input
+★ Imported secret origin key core-20170318210306.
+   hab-studio: Entering Studio at /hab/studios/dev--hab-sln
+** The Habitat Supervisor has been started in the background.
+** Use 'hab svc start' and 'hab svc stop' to start and stop services.
+** Use the 'Get-SupervisorLog' command to stream the supervisor log.
+** Use the 'Stop-Supervisor' to terminate the supervisor.
+
+[HAB-STUDIO] Habitat:\src> build .\hab-sln\
+   : Loading C:\hab\studios\dev--hab-sln\src\hab-sln\\habitat\plan.ps1
+   hab-sln: Plan loaded
+   hab-sln: Validating plan metadata
+   ...
+```
+
+This should compile and package our `hab-sln` project. Lets drill in a bit closer on a few of the moving parts of the plan.
+
+### The nuget command line
+
+Note that our plan takes a build dependency on nuget. That is a core habitat plan that packages the `nuget.exe` utility. Our build will use that to do a "package restore" of all packages declared in our projects `packages.config` file. Visual Studio would essentially do the same if we were building inside the IDE.
```

Comment:
"nuget" => `nuget`

"core habitat plan" => "core Habitat plan"

"... declared in our projects `packages.config`..." => "... declared in our project's `packages.config` ..."